io/ringbuffer.go

```diff
@@ -45,7 +45,7 @@ func (r *RingBuffer) Write(p []byte) (int, error) {
 		r.end++
 		r.size++
 	}
-	return r.size, nil
+	return len(p), nil
```

Comment:
Thanks for the change. Correct me if I am wrong, this change makes `Write` function more language idiomatic  as  it returns the number of bytes written. 

In case of ringBuffer, since it is a revolving buffer, the number of bytes written is always going to be the the length of bytes provided. My intention here was to convey the size of ringbuffer instead. 

If we do this change, may be we should export a `Size()` function on ringbuffer?

Reply:
Sure, probably want the signature to be`Len() int` to match others.